File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int, int> um;
        for(int i=0;i<nums.size();++i) {
            int n = nums[i];
            if(um.find(target-n) != um.end() && i != um[target-n]) return {i, um[target-n]};
            um[n] = i;
        }
        
        return {};
        
    }
};

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = {-1, -1};
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i=0;i<nums.length;i++) {
            if(map.get(nums[i]) != null) {
                ans[0] = map.get(nums[i]);
                ans[1] = i;
                break;
            }
            map.put(target-nums[i], i);
        }
        return ans;
    }
}